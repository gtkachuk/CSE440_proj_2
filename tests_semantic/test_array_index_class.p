program testArrayIndexClass;

class someClass

BEGIN
   VAR irrelevant : integer;
END		  

class testArrayIndexClass

BEGIN

   VAR 
      rows   : ARRAY[0 .. 9] OF boolean;
      tests  : boolean;
      object : someClass;

FUNCTION testArrayIndexClass;
BEGIN	   
   tests := rows[object]
END

END
.

