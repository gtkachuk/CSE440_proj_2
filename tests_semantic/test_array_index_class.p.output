ERROR:20:Array index not an integer for variable "rows"
Errors detected. Exiting.
