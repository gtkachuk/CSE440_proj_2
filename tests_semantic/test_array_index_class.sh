#!/bin/sh

. ../tests_lib.inc

test_diff_sem test_array_index_class.p
