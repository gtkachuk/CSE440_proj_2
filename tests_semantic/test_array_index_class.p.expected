ERROR:21:Array index not an integer for variable "rows"
Errors detected. Exiting.
